fluidkeys/main.go

```diff
@@ -105,23 +106,39 @@ func keySubcommand(args docopt.Opts) exitCode {
 }
 
 func keyFromGpg() exitCode {
+	fluidkeysDirectory, err := getFluidkeysDirectory()
+	if err != nil {
+		fmt.Printf("Failed to get fluidkeys directory: %v\n", err)
+		return 1
+	}
+	db := database.New(fluidkeysDirectory)
 	gpg := gpgwrapper.GnuPG{}
 
-	secretKeys, err := gpg.ListSecretKeys()
+	importedFingerprints, err := db.GetFingerprintsImportedIntoGnuPG()
 	if err != nil {
-		fmt.Errorf("Error getting secret keys from GPG: %v", err)
+		fmt.Printf("Failed to list available keys: %v", err)
 		return 1
 	}
-	fmt.Printf(formatListedKeysForImportingFromGpg(secretKeys))
-	keyToImport := promptForKeyToImportFromGpg(secretKeys)
 
-	fluidkeysDirectory, err := getFluidkeysDirectory()
+	var availableKeys []gpgwrapper.SecretKeyListing
+
+	allGpgKeys, err := gpg.ListSecretKeys()
 	if err != nil {
-		fmt.Printf("Failed to get fluidkeys directory")
+		fmt.Errorf("Error getting secret keys from GPG: %v", err)
```

Comment:
Why's this `Errorf`, and above `Printf`?

Reply:
because i'm a pillock